lib/sidekiq/throttler/rate_limit.rb

```diff
@@ -166,6 +168,8 @@ def executions
 
       private
 
+      SEMAPHORE = Mutex.new
```

Comment:
Can you rename this constant to `LOCK` and move it to the top of our class? That should help clarify its use.